configure.ac

```diff
@@ -2883,8 +2883,36 @@ if test "x$host_win32" = "xyes"; then
     AC_DEFINE(HAVE_CRYPT_RNG)
 fi
 
+AC_MSG_CHECKING(for getrandom)
```

Comment:
Does this really need to be this elaborate? Can't it just be a simple `AC_CHECK_FUNCS(getrandom)`?

Reply:
I'm not 100% sure, but my understanding is that if you have glibc >= 2.25 and Linux kernel older than 3.17, getrandom will return -1 and set errno to ENOSYS and AC_CHECK_FUNCS won't catch that.

Reply:
Then what we should do is detect the `ENOSYS` at runtime and fall back to `/dev/urandom`, IMO.

Reply:
OK, that sounds reasonable.

Can you clarify what `HAVE_CRYPT_RNG` does in `configure.ac`? I just assumed it's required for something, but I can't figure it out.

We probably shouldn't define `HAVE_CRYPT_RNG` if `AC_CHECK_FUNCS` finds `getrandom`, because we don't know at that point if it really works.

Reply:
Perhaps @luhenry could clarify this? I believe he worked on that code.